src/ccs/mod.rs

```diff
@@ -41,15 +41,15 @@ pub struct CCS<C: CurveGroup> {
 impl<C: CurveGroup> CCS<C> {
     /// check that a CCS structure is satisfied by a z vector. Only for testing.
     pub fn check_relation(&self, z: &[C::ScalarField]) -> Result<(), Error> {
-        let mut result = vec![C::ScalarField::zero(); self.m];
+        let mut result = vec![C::ScalarField::zero(); self.n];
```

Comment:
I think this was correct, and lines 77-78 need to be updated to:

```rust
let m = r1cs.A.n_rows;
let n = r1cs.A.n_cols;
```

Reply:
You're right, in CCS the `m` & `n` [docs were swapped](https://github.com/privacy-scaling-explorations/folding-schemes/blob/9ae046c4fc47e1715d6bb2291e8af138b8b836e7/src/ccs/mod.rs#L16), my fault, sorry. Amended fixing it.
Thanks a lot for catching this!

Putting here an example to keep it clear for the future just in case:
Let matrices `Mᵢ` be of size `4 rows x 6 cols` (`m rows x n cols`), we have a `z` (`z=(x,1,w) ∈ Fⁿ` in [CCS paper](https://eprint.iacr.org/2023/552.pdf#section.2)) of size `6`(`=n)`), and we multiply `Mᵢ⋅z` (matrix-vector prod), obtaining a vector of size `4`(`=m`), thus the `hadamard_result` in #52 is of size `m`, and by consequence `result` from line #44 is also of size `m`.
In the test of `ccs/mod.rs`, `m=n_rows=4` and `n=n_cols=6`, and our vector `z` has length `z.len=n=n_cols=6`.